### ISSUE TITLE
Invalid return when jsdoc @abstract is after @return(s)

### ISSUE THREAD

**Issue by dbow** [eslint/eslint#9412]
**Tell us about your environment**

* **ESLint Version:**
v4.7.2

* **Node Version:**
v7.1.0

* **npm Version:**
v3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

module.exports = {
  'extends': 'google',
  'parserOptions': {
    'ecmaVersion': 6
  },
  'rules': {
    'prefer-rest-params': 'off',
    'prefer-spread': 'off',
    'eqeqeq': 'error',
    // Override the google eslint config to prefer @returns
    'valid-jsdoc': [
      'error',
      {
        'requireParamDescription': false,
        'requireReturnDescription': false,
        'requireReturn': false,
        'prefer': {
          'return': 'returns'
        }
      }
    ]
  }
};


```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js

  /**
   * @returns {string}
   * @abstract
   */
 someMethod() {}

```

**What did you expect to happen?**

It [doesn't seem like](https://stackoverflow.com/questions/24306839/is-there-an-official-order-for-jsdoc-tags-in-documentation) there's an expected order for most JSDoc tags? So I would expect order to not matter between the `@abstract` and `@returns` tags. But in the implementation of this [feature request](https://github.com/eslint/eslint/issues/5941) there's an implicit expectation that `@abstract` is *before* `@returns`.

**What actually happened? Please include the actual, raw output from ESLint.**

```
error  Unexpected @returns tag; function has no return statement  valid-jsdoc
```

**Comment by platinumazure** [comment]
Thanks for the report! Seems like a bug to me. We would welcome a pull request to fix this!

**Comment by Orlandster1998** [comment]
@platinumazure I could do that tomorrow.
